Integrations/src/main/java/io/deephaven/integrations/python/PythonReplayListenerAdapter.java

```diff
@@ -87,4 +74,9 @@ public void onUpdate(final TableUpdate update) {
         final boolean isReplay = false;
         pyCallable.call("__call__", update, isReplay);
     }
+
+    @Override
+    public boolean canExecute(final long step) {
+        return source.satisfied(step) && Arrays.stream(dependencies).allMatch(t -> t.satisfied(step));
```

Comment:
I prefer to delegate to `super` for the check on `source`, just in case we add more checks for the normal case.
I also prefer a short-circuiting of the stream ops, no need to pay the allocation costs when there are no extra deps.
```suggestion
        return super.canExecute(step)
                && (dependencies.length == 0 || Arrays.stream(dependencies).allMatch(t -> t.satisfied(step)));
```

Reply:
I had the same thoughts.